test/e2e_node/startup_probe_test.go

```diff
@@ -38,8 +39,15 @@ const (
 )
 
 var _ = framework.KubeDescribe("StartupProbe [Serial] [Disruptive] [NodeFeature:StartupProbe]", func() {
-	f := framework.NewDefaultFramework("critical-pod-test")
+	f := framework.NewDefaultFramework("startup-probe-test")
 	var podClient *framework.PodClient
+	ginkgo.BeforeEach(func() {
+		// It's not enough to set this flag in the kubelet because the apiserver needs it too
+		if !utilfeature.DefaultFeatureGate.Enabled(features.StartupProbe) {
```

Comment:
this makes it impossible to run this test until the feature is promoted out of alpha. I'd expect the `NodeFeature:StartupProbe` to omit by default but allow specific CI setups that enable the feature across the apiserver and kubelet to opt into testing it

Reply:
Yes, if only enabled by kubelet worker, it's not meaningful.

Reply:
> it's not meaningful for the e2e process to have the feature gate enabled

this was my main point... and this PR is still trying to change behavior based on whether the feature is enabled in the e2e process

Reply:
@liggitt I don't understand what should be changed here to successfully run the test from the main CI... could you help me fix that test?

Reply:
Normally, tests that depend on features that are not on by default add something like `[NodeFeature:StartupProbe]` or `[Feature:StartupProbe]` to their description (as this already did).

By default, CI invocations typically skip tests with `[Feature:[^]]*]` or `[NodeFeature:[^]]*]` (using `ginkgo.skip` flags, like https://github.com/kubernetes/test-infra/blob/0fb5415c9822d2f4b8de256c8c2d5ce943f5617e/config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml#L53)

If a particular CI job knows it is enabling a non-default feature (in the way it is starting up the kubelet or kube-apiserver, for example), it can add back in the tests (using ginkgo.focus flags, like https://github.com/kubernetes/test-infra/blob/913a03e8743180335dd34f5301f8ffafe711d5e9/config/jobs/gke/containerd/gke-test-containerd.yaml#L94)

Reply:
It seems like you need to do the following:
* identify the failing test jobs and ensure their invocations are skipping `[NodeFeature:StartupProbe]`
* figure out what test job is exercising alpha node features and make sure this test is running/passing in that job

Reply:
That makes sense. So if I understand this correctly, the test should be marked as `NodeAlphaFeature` instead of the current `NodeFeature` in order to skip it in the default kubelet-serial test, since it is still in alpha.


However, the line `podClient = f.PodClient()` is still required since the test currently panics because it is lacking.

Reply:
Damn, I let that slip with the panic before codefreeze...
@zouyee do you have time to change your PR accordingly or should I open another one?

Reply:
@zouyee as I understand it, with `NodeAlphaFeature:StartupProbe` you no longer need to skip depending on the feature flag...

Reply:
@matthyx  I close it, please post other pr, thanks 😄